spec/server/listener_metrics_spec.rb

```diff
@@ -223,8 +295,61 @@ def authrep_query(args)
         '/transactions/authrep.xml?' + parsed_query_args(args)
       end
 
-      def get_service_internal_api(listener_host, listener_port, service_id)
-        Net::HTTP.get(listener_host, "/internal/services/#{service_id}", listener_port)
+      def do_internal_api_get_req(path)
+        do_get_req("/internal#{path}")
+      end
+
+      def do_internal_api_delete_req(path)
+        do_delete_req("/internal#{path}")
+      end
+
+      def get_service_internal_api(service_id)
+        do_internal_api_get_req("/services/#{service_id}")
+      end
+
+      def get_alerts_internal_api(service_id)
+        do_internal_api_get_req("/services/#{service_id}/alert_limits/")
+      end
+
+      def get_app_keys_internal_api(service_id, app_id)
+        do_internal_api_get_req("/services/#{service_id}/applications/#{app_id}/keys/")
+      end
+
+      def get_app_referrer_filters_internal_api(service_id, app_id)
+        do_internal_api_get_req("/services/#{service_id}/applications/#{app_id}/referrer_filters")
+      end
+
+      def get_app_internal_api(service_id, app_id)
+        do_internal_api_get_req("/services/#{service_id}/applications/#{app_id}")
+      end
+
+      def get_errors_internal_api(service_id)
+        do_internal_api_get_req("/services/#{service_id}/errors/")
+      end
+
+      def get_events_internal_api
+        do_internal_api_get_req("/events/")
+      end
+
+      def get_metric_internal_api(service_id, metric_id)
+        do_internal_api_get_req("/services/#{service_id}/metrics/#{metric_id}")
+      end
+
+      def get_service_token_internal_api(token, service_id)
+        do_internal_api_get_req("/service_tokens/#{token}/#{service_id}/provider_key")
+      end
+
+      def delete_stats_internal_api(service_id)
+        do_internal_api_delete_req("/services/#{service_id}/stats")
+      end
+
+      def get_usage_limits_internal_api(service_id, plan_id, metric_id, period)
+        do_internal_api_get_req("/services/#{service_id}/plans/#{plan_id}/usagelimits/" +
+                                "#{metric_id}/#{period}")
+      end
+
+      def get_utilization_internal_api(service_id, app_id)
+        do_internal_api_get_req("/services/#{service_id}/applications/#{app_id}/utilization/")
```

Comment:
What a magnificient opportunity to generate these programmatically by iterating a hashmap with method names and paths with placeholders for arguments. :D

... but now it's done and we wouldn't gain anything from it at this point, so :+1:.